File: Task-002/HDFSwordcount.scala
import scala.io.Source
import java.io.File
import java.io.PrintWriter
import org.apache.hadoop.conf.Configuration
import org.apache.hadoop.fs.FileSystem
import org.apache.hadoop.fs.Path

def wordCount (substr: String): Int =
  wordsAll.sliding(substr.length).count(window => window == substr)

def main(args: Array[String]) {
  val wordsAll = Source.fromFile("Shakespeare.txt").getLines.mkString.replace("\n", "").replaceAll("""[\p{Punct}]""", " ").toLowerCase()

  val wordsSet: Set[String] = wordsAll.split(" ").toSet

  val writer = new PrintWriter(new File("shakespeareMR.txt", false))

  for (x <- wordsSet) {
    var n = wordCount(x)
    writer.write(s"$x: $n\n")
  }

  writer.close()

  val hadoopConf = new Configuration()
  val hdfs = FileSystem.get(hadoopConf)

  val srcPath = new Path("shakespeareMR.txt")
  val destPath = new Path("/shakespeare/shakespeareMR.txt")

  hdfs.copyFromLocalFile(srcPath, destPath)
}

// Error: Could not find or load main class scala.tools.nsc.MainGenericRunner
// It was working, then the laptop died and now it's not working.

File: Task-002/HDFSwordcount.py
import string
import os
from subprocess import PIPE, Popen

with open("Shakespeare.txt", "r") as f:
	words_all = f.read().replace("\n", "").lower()

for ch in string.punctuation:
	words_all = words_all.replace(ch, " ")

words_set = words_all.split(" ")

t = open("shakespeareMR.txt","w+")

for x in words_set:
	n = words_all.count(x)
	t.write("{}: {}\n".format(x,n))

t.close()

hdfs_path = os.path.join(os.sep, "shakespeare", "shakespeareMR.txt")

put = Popen(["hdfs", "dfs", "-put", "-f", "shakespeareMR.txt", hdfs_path], stdin = PIPE, bufsize = -1)
put.communicate

## End the HDFS terminal screen?
